core/src/main/java/io/grpc/internal/TransportFrameUtil.java

```diff
@@ -61,7 +61,8 @@
       if (endsWith(key, binaryHeaderSuffixBytes)) {
         // Binary header.
         serializedHeaders[k] = key;
-        serializedHeaders[k + 1] = BaseEncoding.base64().encode(value).getBytes(US_ASCII);
+        serializedHeaders[k + 1]
+            = BaseEncoding.base64().omitPadding().encode(value).getBytes(US_ASCII);
```

Comment:
This will now do an allocation. Let's configure the base encoder once outside of the for loop.

Reply:
Can I declare a public static final instance in this class, and use it everywhere (except for core)?

Reply:
Now I use an static instance in `InternalMetadata`.